### PR TITLE
Implementation For Transfer Related APIs

### PR DESCRIPTION
**Pull request by wharleyinc**
##Implementation

Added models for all transfer related data objects
Added methods in the RemoteCall interface to facilitate transfer related requests
Added an example package to test remote calls

[https://docs.ovalfi.com/docs/initiate-transfer](url)
##Paired With
CC @odetolakehinde

### CODE REVIEW

File: api/transfer.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"github.com/ovalfi/go-sdk/helpers"
+	"github.com/ovalfi/go-sdk/model"
+	"net/http"
```
Comment (odetolakehinde) [inline api/transfer.go:8]: please sort you imports.

File: api/transfer.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// InitiateTransfer makes an API request using Call to initiate a transfer
+func (c *Call) InitiateTransfer(ctx context.Context, request model.InitiateTransferRequest) (model.Transfer, error) {
+	endpoint := fmt.Sprintf("%s%s", c.baseURL, transferAPIVersion)
+
+	fL := c.logger.With().Str("func", "InitiateTransfer").Str("endpoint", endpoint).Logger()
+	fL.Info().Msg("starting...")
+	fL.Info().Str("customerID", request.CustomerID).Str("reference", request.Reference).
+		Str("currency", request.Currency).Interface("destination", request.Destination).
+		Float64("amount", request.Amount).Str("note", request.Note).Str("reason", request.Reason).
+		Str("reference", request.Reference).
```
Comment (odetolakehinde) [inline api/transfer.go:22]: you can just do `.Interface("request", request)`

File: model/model.go [hunk 3]
```diff
@@ -1,7 +1,11 @@
 // Package model defines object and payload models
 package model
 
-import "database/sql"
+import (
+	"database/sql"
+	"github.com/google/uuid"
+	"time"
```
Comment (odetolakehinde) [inline model/model.go:7]: 👀

File: api/transfer.go (hunk 2 above)
Comment (wharleyinc) [reply api/transfer.go:22]: Updated

**Review by odetolakehinde — APPROVED** [review]

**Review by odetolakehinde — APPROVED** [review]

**Review by azeezlala — APPROVED** [review]

**Pull request merged**